apps/designer/app/shared/db/assets.server.ts

```diff
@@ -15,11 +18,30 @@ export const loadByProject = async (projectId?: Project["id"]) => {
 
 export const create = async (
   projectId: Project["id"],
-  values: { type: string; name: string; path: string }
+  values: { name: string; path: string },
+  absolutePath?: string
 ) => {
+  let size = 0;
+  const image = await sharp(absolutePath);
+  const metadata = await image.metadata();
```

Comment:
what are  perf characteristics of those calls?

Reply:
we can make a naive bench by running this in a for loop 1000 times and measure average time spent there

Reply:
https://sharp.pixelplumbing.com/performance#the-task

Reply:
also what if upload was to s3, will those download again to get meta?

Reply:
> https://sharp.pixelplumbing.com/performance#the-task

that's a bit too generic, what's the cost of those specific calls is the question

Reply:
and also how can you do it if you don't have a local file

Reply:
yeah because idk why remix doesn't send the file size or the actual file after doing the file upload, at least for now.

Will see what happens with the other one, I think I have to make my own handler so I can make sure to still return the image array buffer and I can pass that to sharp

Reply:
there might be a way to get this out of remix, check out their implementation

Reply:
at least the size

Reply:
Regarding metadata, if this lib can only make like 30 images per second, its a nogo, way too slow to be in the main server process. Also if we have to upload to S3, you would have to redownload. 

Instead I would try looking into approaches which can take a few bytes and read exif data from there, without downloading the full image or needing access directly to the file system, this way we can get those bytes from any storage and parse the exif data

Reply:
For context: image exif data is always first few bytes of the image, but majority of graphic tools read the entire image because they are preparing to do more stuff with the image. I used a tiny parser many years ago, it will operate in microseconds area.

Reply:
ignore this, it uses the array buffer now